crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +140,133 @@ impl Persistence {
             .await
             .map_err(Error::DbError)
     }
+
+    /// Get auction data.
+    pub async fn get_auction(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::settlement::Auction, error::Auction> {
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .context("begin")
+            .map_err(error::Auction::DbError)?;
+
+        let deadline = {
+            let scores = database::settlement_scores::fetch(&mut ex, auction_id)
+            .await
+            .context("fetch scores").map_err(error::Auction::DbError)?
+            // if score is missing, no competition / auction exist for this auction_id
+            .ok_or(error::Auction::Missing)?;
+
+            (scores.block_deadline as u64).into()
+        };
+
+        let prices = {
+            let db_prices = database::auction_prices::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction prices")
+                .map_err(error::Auction::DbError)?;
+
+            let mut prices = HashMap::new();
+            for price in db_prices {
+                let token = eth::H160(price.token.0).into();
+                let price = big_decimal_to_u256(&price.price)
+                    .ok_or(domain::auction::InvalidPrice)
+                    .and_then(|p| domain::auction::Price::new(p.into()))
+                    .map_err(error::Auction::Price)?;
+                prices.insert(token, price);
+            }
+            prices
+        };
+
+        let orders = {
+            // get all orders from a competition auction
+            let auction_orders = database::auction_orders::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction orders")
+                .map_err(error::Auction::DbError)?
+                .ok_or(error::Auction::Missing)?
+                .into_iter()
+                .map(|order| domain::OrderUid(order.0))
+                .collect::<HashSet<_>>();
+
+            // get quotes for all auction orders
+            let quotes = self
+                .postgres
+                .read_quotes(auction_orders.iter())
+                .await
+                .map_err(error::Auction::DbError)?;
```

Comment:
As Felix [suggested in a separate PR](https://github.com/cowprotocol/services/pull/2809#discussion_r1679132985), let's not fetch all the quotes. Here is how I tried to address it: https://github.com/cowprotocol/services/pull/2809/files#diff-ba91e794defad0d0d318c65a444c3bd6bfc1c6cee8abc31c5af7eb07ae428093R24-R48

Reply:
fixed